server/routes/mpesa.py

```
@@ -0,0 +1,419 @@
+
+            else:
+                transaction.status = "failed"
+
+            db.session.commit()
+            return jsonify({"ResultCode": 0, "ResultDesc": "Success"})
+
+        return jsonify({"ResultCode": 1, "ResultDesc": "Invalid callback format"}), 400
+
+    except Exception as e:
+        db.session.rollback()
+        current_app.logger.exception(f"Error processing callback {e}")
```

Ruff (0.13.1)

Redundant exception object included in `logging.exception` call

(TRY401)